jupyter_dashboards/nbextension/notebook/dashboard-view/dashboard-actions.js

```diff
@@ -2,131 +2,138 @@
+    var currentState = STATE.NOTEBOOK;
+    var toolbarBtnsSelector = '#urth-dashboard-view-toolbar-buttons';
+    var isHeaderVisible = true, isToolbarVisible = true;
+    var opts, scrollToBottom;
+
+    function enterDashboardMode(newState) {
+        opts.enterDbModeCallback(
+            newState !== STATE.DASHBOARD_VIEW, // doEnableGrid
+            newState === STATE.DASHBOARD_AUTH_REPORT
+        );
+        // disable scroll to bottom of notebook
+        scrollToBottom = IPython.Notebook.prototype.scroll_to_bottom;
+        IPython.Notebook.prototype.scroll_to_bottom = function() {};
     }
 
     function exitDashboardMode() {
-        // Revert changes done in enterDashboardMode()
-        $('body').removeClass('view-only');
-        $('#urth-notebook-view').addClass('selected');
-        $('#urth-dashboard-auth, #urth-dashboard-view').removeClass('selected');
-        $('#urth-dashboard-show-all, #urth-dashboard-show-all-stacked, #urth-dashboard-hide-all').addClass('disabled');
-        exitDbModeCallback();
-        updateUrlState(false);
-        toggleHeaders(true); // enable header and toolbar
-        enableCodeMirror(true);
-
+        opts.exitDbModeCallback();
         // restore scrolling behavior
         IPython.Notebook.prototype.scroll_to_bottom = scrollToBottom;
         scrollToBottom = null;
     }
 
-    function enterDashboardMode(newState, prevState) {
-        // Add a class to the document body so our styles can take effect
-        $('body').toggleClass('view-only', newState === STATE_DASHBOARD_VIEW);
-
-        // style menus accordingly
-        $('#urth-notebook-view').removeClass('selected');
-        $('#urth-dashboard-auth').toggleClass('selected', newState === STATE_DASHBOARD_AUTH);
-        $('#urth-dashboard-view').toggleClass('selected', newState === STATE_DASHBOARD_VIEW);
-        $('#urth-dashboard-show-all, #urth-dashboard-show-all-stacked, #urth-dashboard-hide-all').toggleClass('disabled', newState === STATE_DASHBOARD_VIEW);
-
-        // disable header and toolbar as necessary
-        toggleHeaders(newState === STATE_DASHBOARD_AUTH);
-
-        enableCodeMirror(false);
-
-        enterDbModeCallback(newState === STATE_DASHBOARD_AUTH /* doEnableGrid */);
-        updateUrlState(newState === STATE_DASHBOARD_VIEW);
-
-        // disable scroll to bottom of notebook
-        scrollToBottom = IPython.Notebook.prototype.scroll_to_bottom;
-        IPython.Notebook.prototype.scroll_to_bottom = function() {};
+    function getFaIconClass(el) {
+        return $(el).attr('class').split(' ').filter(function(value) {
+            return /^fa-/.test(value);
+        })[0];
     }
 
-    function setViewButtonEnabled(state) {
-        var idx = VIEW_BTN_IDX[state];
-        $('#urth-dashboard-view-toolbar-buttons > button:nth-of-type(' + idx + ')')
-            .addClass('active')
-            .siblings().removeClass('active');
+    function updateAuthoringBtn(state) {
+        if (/^db-auth-/.test(state)) {
```

Comment:
If we're setting up (basically) an ENUM at the beginning of the file (`var STATE = {...}`), I think we should use that everywhere, rather than sometimes also using the values. I know it will be more verbose, but it will make it easier to maintain and search in the future.

If you really need to map from values to enum, that should be refactored into its own getter function, preferably right below the definition of the enum itself.

(Here and elsewhere).